app/controllers/api/v1/job_applications_controller.rb

```diff
@@ -11,6 +11,17 @@ def create
     end
   end
 
+  def show
+    user = User.find(params[:user_id])
+    job_application = JobApplication.find_by(id: params[:id])
+
+    if job_application.nil? || job_application.user_id != user.id
```

Comment:
Would we want a different error if the user doesn't match?

Reply:
Kinda torn on this, Im not sure if the effort is worth the result, but i'm willing to hear out justifications. Feel free to post some in this thread if anyone is interested.

Reply:
Not 100% one way or the other, but clearer messages to the front is helpful and since a lack of job application is a slightly different error than an incorrect user id. They both result in the job application not working as intended, so I also see why one error for each does the job.